plugins/UsersManager/Model.php

```diff
@@ -245,11 +245,14 @@ public function userEmailExists($userEmail)
     public function addUserAccess($userLogin, $access, $idSites)
     {
         foreach ($idSites as $idsite) {
-            $this->getDb()->insert(Common::prefixTable("access"),
-                array("idsite" => $idsite,
-                      "login"  => $userLogin,
-                      "access" => $access)
-            );
+            $count = $this->getDb()->fetchOne("SELECT count(*) FROM " . Common::prefixTable("site"). " WHERE idsite = ?",$idsite);
+            if($count != 0){
+                 $this->getDb()->insert(Common::prefixTable("access"),
+                    array("idsite" => $idsite,
+                         "login"  => $userLogin,
+                         "access" => $access)
+                );
+            }
```

Comment:
I am not.

On Tue, Nov 10, 2015, 12:42 AM Thomas Steur <notifications@github.com>
wrote:

> In plugins/UsersManager/Model.php
> <https://github.com/piwik/piwik/pull/8718#discussion_r44316670>:
>
> > @@ -245,11 +245,14 @@ public function userEmailExists($userEmail)
> >      public function addUserAccess($userLogin, $access, $idSites)
> >      {
> >          foreach ($idSites as $idsite) {
> > -            $this->getDb()->insert(Common::prefixTable("access"),
> > -                array("idsite" => $idsite,
> > -                      "login"  => $userLogin,
> > -                      "access" => $access)
> > -            );
> > +            $count = $this->getDb()->fetchOne("SELECT count(*) FROM " . Common::prefixTable("site"). " WHERE idsite = ?",$idsite);
> > +            if($count != 0){
> > +                 $this->getDb()->insert(Common::prefixTable("access"),
> > +                    array("idsite" => $idsite,
> > +                         "login"  => $userLogin,
> > +                         "access" => $access)
> > +                );
> > +            }
>
> Hey @haseebeqx <https://github.com/haseebeqx> are you still working on
> the PR? Can I maybe help you somehow or explain it in a different way?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/piwik/piwik/pull/8718/files#r44316670>.
>